app/controllers/users_controller.rb

```diff
@@ -34,6 +36,20 @@ def destroy
   def index
   end
   
+  # Confirms a logged-in user.
+  def logged_in_user
+    unless logged_in?
+      flash[:danger] = "Please log in."
+      redirect_to login_path
+    end
+  end
+  
+  # Confirms the correct user.
+  def correct_user
+    @user = User.find(params[:id])
```

Comment:
@kevinchen93, the nuance is slightly different. I can input any id param in the url, so this function basically checks that the currently logged in user is equivalent to the id param passed in.